examples/tensorflow/tf2/train_predict_2.py

```diff
@@ -63,7 +63,7 @@ def eval_input_fn(features, labels, batch_size):
 
 
 # Enable auto-logging to MLflow to capture TensorBoard metrics.
-mlflow.tensorflow.autolog()
+mlflow.tensorflow.autolog(log_model_signatures=False)
```

Comment:
Why do we need to disable signature logging for this example?

Reply:
Good point. When `pyfunc_model` is loaded in this example, it is predicted with `pd.DataFrame` and the predictions are validated against that type as well, but if `log_model_signatures` is set to `True`, then the output is always of predict is always of type numpy.ndarray (or dict ) and so the example logic breaks. I thought it was just simpler to turn `log_model_signatures` off for this example, but let me know if you would like the example to be adjusted rather?!

Reply:
Interesting! In that case, the fact that automatically including a signature changes the output type of `predict()` constitutes a backwards-incompatible change. If a user was running the workflow from this example in production using MLflow 1.24 and then upgraded to a version of MLflow that contains the changes made in this PR, their workflow would break.

Can we root cause why the output type is changing and see if we can fix or work around it?

Reply:
In particular, it's a bit curious to me that the output of `predict()` with a Pandas DataFrame input (as provided by this example - https://github.com/mlflow/mlflow/pull/5510/files#diff-3ca06392b52de110cff4856f4cd144a4fdc801fba93dce851aaa7f0d66993c79R153-R160) would produce anything other than a DataFrame, given the following logic: https://github.com/mlflow/mlflow/blob/2e175e7fd65d86d12d204b648aa5ff73063a7dc9/mlflow/tensorflow/__init__.py#L533

Reply:
With`log_model_signatures` set to `True`, the data is enforced as 
```
{'SepalLength': array([5.1, 5.9, 6.9]), 'SepalWidth': array([3.3, 3. , 3.1]), 'PetalLength': array([1.7, 4.2, 5.4]), 'PetalWidth': array([0.5, 1.5, 2.1])}```

by [_enforce_schema](https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/__init__.py#L622)
to be of type dict (str -> ndarray), which actually hits [this](https://github.com/mlflow/mlflow/blob/2e175e7fd65d86d12d204b648aa5ff73063a7dc9/mlflow/tensorflow/__init__.py#L531)...i think to be able to put a backward compatibility fix we need to know the original type passed to the pyfunc model predict, which is lost when it gets to the TF flavor's [predict](https://github.com/mlflow/mlflow/blob/2e175e7fd65d86d12d204b648aa5ff73063a7dc9/mlflow/tensorflow/__init__.py#L503).

Reply:
Isn't the original data submitted to that predict method in the TF Wrapper? that explicit dict casting of {column_class: <values>} is performed within that predict method (line 524::528). 
Why is the type validation in line 530 returning a dict and not a pandas DataFrame from line 533? Shouldn't it be re-casting to a DF?

Reply:
Thanks @dbczumar @BenWilson2 ! Will put those fixes shortly including all open comments below.

Reply:
Committed, please let me know if you would like further changes to it.